### PR TITLE
Elasticsearch: Add settings to switch between new and legacy table model for Raw Documents

### LINKED ISSUE #24163: Elasticsearch: Raw document query render raw JSON documents in each row in the new table panel
**What happened**:
Raw document query render raw JSON documents in each row in the new table panel.

image

**What you expected to happen**:
Each property in raw JSON to be displayed as a column as with the "old table panel".

**How to reproduce it (as minimally and precisely as possible)**:
1. Go to https://play.grafana.org/d/000000014/elasticsearch-metrics?orgId=1&editPanel=5
2. Switch to the "new table panel"
3. Table shows raw JSON documents in each row

**Environment**:
- Grafana version: v7.0.0-beta1
- Data source type & version: Elasticsearch

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
Following up on our discussion in #26233, I have added a switch to handle Series/DataFrame table model for Raw Document Elastic response. 

Kapture 2020-07-14 at 12 57 41

**Which issue(s) this PR fixes**:
Fixes #24163

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/elastic_response.ts [hunk 1]
```diff
@@ -391,72 +391,103 @@ export class ElasticResponse {
   }
 
   getTimeSeries() {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/elastic_response.ts:393]: no change here compared to https://github.com/grafana/grafana/pull/26233

File: public/app/plugins/datasource/elasticsearch/query_builder.ts [hunk 2]
```diff
@@ -212,6 +212,7 @@ export class ElasticQueryBuilder {
     // If target doesn't have bucketAggs and type is not raw_document, it is invalid query.
     if (target.bucketAggs.length === 0) {
       metric = target.metrics[0];
+
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/query_builder.ts:215]:
```suggestion
```

File: public/app/plugins/datasource/elasticsearch/elastic_response.ts [hunk 3]
```diff
@@ -391,72 +391,103 @@ export class ElasticResponse {
...
-    const seriesList = [];
+    if (
+      this.targets.some((target: any) => target.metrics.some((metric: any) => metric.settings?.useNewFormat === true))
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/elastic_response.ts:395]: this is the only line that changed in `public/app/plugins/datasource/elasticsearch/elastic_response.ts` compared to https://github.com/grafana/grafana/pull/26233

**Comment by ivanahuckova** [comment]
Hello!👋 I have couple questions:
- @annelaurefroment, you have mentioned that this fix was also for our enterprise customer and that it should be part of 7.1. Would this switch in Elastic query editor work as a solution for the problem? 
- @torkelo @marefr, I have talked to @aocenas and we were trying to figure out how should the default setting `useNewFormat = true` work and how would we handle the fact, that opening an edit mode to the old table panel would automatically change the visualisation in table, as the `useNewFormat` would be set to true. We were also thinking about setting default to yes for new panels and no for old panels, but this wouldn't be ideal, as it would still change the table visualisation for some of our customers. Therefore, we have agreed on setting `useNewFormat = false` and if users wan't to see individual fields, they can do it with 1 click on the switch. This way, we are not breaking current behaviour of older panels and we are giving users ability to set their preferred table view. Also, we are not changing anything and therefore we have time to figure out/come out with better idea on how to decide when to set `useNewFormat === true`.

**Comment by torkelo** [comment]
Why can't it default to true for new queries? Think a old suggested solution of a new raw data metric option sounds better as for that we do not need to introduce a new bool property whose value is not what we want (it would be ok if it was useLegacy and defaulted to false , but that is not possible) a useNewFormat that is always needs to be true forever feels just messy. But not sure which solution is best both are a bit messy 🤔

**Comment by ivanahuckova** [comment]
>Why can't it default to true for new queries?

Well, we just thought about the scenario, where you open a previously created table panel in edit mode. This would automatically change the visualisation in table, as the useNewFormat would be set to true. So not sure if that would be something we want. Maybe?

>Think a old suggested solution of a new raw data metric option sounds better... But not sure which solution is best both are a bit messy 

Yeah, honestly not sure either.

**Comment by annelaurefroment** [comment]
@ivanahuckova thanks, yes I believe this would work for them. They would prefer a solution where they do not have to make changes to their existing dashboards, but just a switch like that should be OK.

**Comment by marefr** [comment]
Agree with @torkelo that it gets messy with new property that is true by default. Sorry for the inconvenience. Lets close this and continue with #26233 then.

**Pull request closed without merging**